### ISSUE TITLE
Naming convention for gradient or jacobian variables and methods

### ISSUE THREAD

**Issue by gilrrei** [queens-py/queens#24]
> This issue was originally created by @jnitzler on [gitlab.lrz.de](https://gitlab.lrz.de/queens_community/queens/-/issues/360) on 2022-06-27.

## Motivation and Context

At the moment we have different naming strategies for gradients and jacobians. Sometimes the docstring elaborates with respect to which variable or function a gradient is taken and sometimes we try to express the latter by the name of the variable itself.

For more complex variations/gradients/jacobians variable names might get lengthy. It would be nice to have a unified strategy in queens to avoid mistakes.

## Related Issues / MR

* Blocks
* Is blocked by
* Follows [342](https://gitlab.lrz.de/queens_community/queens/-/merge_requests/342) 
* Precedes
* Related to
* Part of
* Composed of

## Interested Parties

@gilrrei @maxdinkel
